code/obj/mining.dm

```diff
@@ -378,8 +378,28 @@
 				do_malfunction()
 			sleep(sleep_time)
 
-			var/datum/mining_encounter/MC = get_encounter(rarity_mod)
-			MC.generate(target)
+			var/datum/mining_encounter/MC
+
+			if(selectable_encounter_id != null)
+				if(mining_controls.mining_encounters_selectable.Find(selectable_encounter_id))
```

Comment:
```suggestion
				if(selectable_encounter_id in mining_controls.mining_encounters_selectable)
```

Reply:
Should that change also be applied to line 619? I coped the code block from there.

Reply:
yes